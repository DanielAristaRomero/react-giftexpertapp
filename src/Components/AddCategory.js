import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategorias}) => {

    const [inputValue, setInputValue] = useState('');

    const handelInputChange = (e) => {
        setInputValue(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();

        if(inputValue.trim().length > 2) {
            setCategorias(cats => [inputValue, ...cats]);
            setInputValue('');
        }

        
    };

    return (
        <form onSubmit = {handleSubmit}>
            <input 
                className = 'buscador'
                type = 'text'
                value = {inputValue}
                onChange = {handelInputChange}
                />
        </form>
    );
}

AddCategory.propTypes = {
    setCategorias: PropTypes.func.isRequired,
}