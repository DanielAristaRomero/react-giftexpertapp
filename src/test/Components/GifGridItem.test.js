import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../Components/GifGridItem';


describe('Pruebas en <GifGridItems />', () => {
    
    const title = 'Titulo';
    const url = 'http://localhost/algo.png';
    const wrapper = shallow(<GifGridItem title = {title} url = {url}/>);

    test('debe de mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('debe de tener un pÃ¡rrafo con el title', () => {
       const p = wrapper.find('p');
       expect(p.text().trim()).toBe(title); 
    })

    test('debe de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })
    
    test('debe de tener animate__bounce', () => {
        const clase = 'animate__bounce';
        const div = wrapper.find('div');
        expect(div.prop('className').includes(clase)).toBe(true);

    })
    
    
});
