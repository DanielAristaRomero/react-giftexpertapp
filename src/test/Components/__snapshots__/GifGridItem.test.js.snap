// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItems /> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounceIn"
>
  <img
    alt="Titulo"
    src="http://localhost/algo.png"
  />
  <p>
    Titulo
  </p>
</div>
`;
